openedx/core/djangoapps/user_authn/views/password_reset.py

```diff
@@ -299,171 +300,222 @@ def _uidb36_to_uidb64(uidb36):
+            return response
+
+    def _handle_primary_email_update(self, updated_user):
+        try:
+            updated_user.email = updated_user.account_recovery.secondary_email
+            updated_user.account_recovery.delete()
+            # emit an event that the user changed their secondary email to the primary email
+            tracker.emit(
+                SETTING_CHANGE_INITIATED,
+                {
+                    "setting": "email",
+                    "old": self.user.email,
+                    "new": updated_user.email,
+                    "user_id": updated_user.id,
+                }
             )
+        except ObjectDoesNotExist:
+            log.error('Account recovery process initiated without AccountRecovery instance for user {username}'
+                      .format(username=updated_user.username))
+
+    def _handle_password_creation(self, request, updated_user):
+        messages.success(
+            request,
+            HTML(_(
+                u'{html_start}Password Creation Complete{html_end}'
+                u'Your password has been created. {bold_start}{email}{bold_end} is now your primary login email.'
+            )).format(
+                support_url=configuration_helpers.get_value('SUPPORT_SITE_LINK', settings.SUPPORT_SITE_LINK),
+                html_start=HTML('<p class="message-title">'),
+                html_end=HTML('</p>'),
+                bold_start=HTML('<b>'),
+                bold_end=HTML('</b>'),
+                email=updated_user.email,
+            ),
+            extra_tags='account-recovery aa-icon submission-success'
+        )
+
+    def post(self, request, *args, **kwargs):
+        # We have to make a copy of request.POST because it is a QueryDict object which is immutable until copied.
+        # We have to use request.POST because the password_reset_confirm method takes in the request and a user's
+        # password is set to the request.POST['new_password1'] field. We have to also normalize the new_password2
+        # field so it passes the equivalence check that new_password1 == new_password2
+        # In order to switch out of having to do this copy, we would want to move the normalize_password code into
+        # a custom User model's set_password method to ensure it is always happening upon calling set_password.
+        request.POST = request.POST.copy()
+        request.POST['new_password1'] = normalize_password(request.POST['new_password1'])
+        request.POST['new_password2'] = normalize_password(request.POST['new_password2'])
+
+        password = request.POST['new_password1']
+        response = self._validate_password(password, request)
+        if response:
+            return response
 
-        # If password reset was unsuccessful a template response is returned (status_code 200).
-        # Check if form is invalid then show an error to the user.
-        # Note if password reset was successful we get response redirect (status_code 302).
```

Comment:
Could you preserve this comment in the updated code?  Without this context, it looks bizarre for a 200 response to be treated as a password reset failure.

Reply:
done